src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -85,11 +85,48 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
             override fun visitAttr(o: RsAttr) = checkAttr(holder, o)
             override fun visitRangeExpr(o: RsRangeExpr) = checkRangeExpr(holder, o)
             override fun visitTraitType(o: RsTraitType) = checkTraitType(holder, o)
+            override fun visitPatStruct(o: RsPatStruct) = checkRsPatStruct(holder, o)
+            override fun visitPatTupleStruct(o: RsPatTupleStruct) = checkRsPatTupleStruct(holder, o)
         }
 
         element.accept(visitor)
     }
 
+    private fun checkRsPatStruct(holder: AnnotationHolder, rsPatStruct: RsPatStruct) {
+        val declaration = rsPatStruct.path.reference.deepResolve() as? RsFieldsOwner ?: return
+        val declarationFieldNames = declaration.fields.map { it.name }
+        val bodyFields = rsPatStruct.patFieldList
+        val extraFields = bodyFields.filter { it.kind.fieldName !in declarationFieldNames }
+        val bodyFieldNames = bodyFields.map { it.kind.fieldName }
+        val missingFields = declaration.fields.filter { it.name !in bodyFieldNames }
+        if (extraFields.isNotEmpty()) {
+            RsDiagnostic.ExtraFieldsInStructPattern(rsPatStruct, extraFields, missingFields.isEmpty()).addToHolder(holder)
+        }
+        if (missingFields.isNotEmpty() && rsPatStruct.dotdot == null) {
+            if (declaration.elementType == RsElementTypes.ENUM_VARIANT) {
+                RsDiagnostic.MissingFieldsInEnumVariantPattern(rsPatStruct, declaration).addToHolder(holder)
+            } else {
+                RsDiagnostic.MissingFieldsInStructPattern(rsPatStruct, declaration).addToHolder(holder)
+            }
+        }
+    }
+
+    private fun checkRsPatTupleStruct(holder: AnnotationHolder, rsPatTupleStruct: RsPatTupleStruct) {
+        val declaration = rsPatTupleStruct.path.reference.deepResolve() as? RsFieldsOwner ?: return
+        val bodyFields = rsPatTupleStruct.childrenOfType<RsPatIdent>()
+        if (bodyFields.size < declaration.fields.size) {
+            if (declaration.elementType == RsElementTypes.ENUM_VARIANT) {
+                RsDiagnostic.MissingFieldsInEnumVariantPattern(rsPatTupleStruct, declaration).addToHolder(holder)
+            } else {
+                RsDiagnostic.MissingFieldsInStructPattern(rsPatTupleStruct, declaration).addToHolder(holder)
+            }
+        } else if (bodyFields.size > declaration.fields.size) {
```

Comment:
```suggestion
        } else if (bodyFields.size > declaration.fields.size && rsPatTupleStruct.dotdot == null) {
```

Tuple patterns can also contain `..`:
```rust
let Foo(a, .., b) = Foo(...);
```